lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -317,32 +317,53 @@ def increment_positions_on_all_items
         # Reorders intermediate items to support moving an item from old_position to new_position.
         def shuffle_positions_on_intermediate_items(old_position, new_position, avoid_id = nil)
           return if old_position == new_position
-          scope = acts_as_list_list
+          scope = acts_as_list_list.select(:id)
 
           if avoid_id
             scope = scope.where("#{quoted_table_name}.#{self.class.primary_key} != ?", self.class.connection.quote(avoid_id))
           end
 
+          # unique constraint prevents regular increment_all and forces to do increments one by one
+          # http://stackoverflow.com/questions/7703196/sqlite-increment-unique-integer-field
+          # both SQLite and PostgreSQL (and most probably MySQL too) has same issue
+          update_one_by_one = acts_as_list_list.connection.index_exists?(acts_as_list_list.table_name, position_column, unique: true)
```

Comment:
At first I thought that if you have unique index this is not an option, really, but a requirement. Still someone may want to disable this for performance reasons iif DB has deferred unique check on this table, and this will work too. Preparing update to make :sequential_updates option.